File: daily/Dec25.cc
#include <iostream>
using namespace std;

class Solution {
 public:
  int64_t minimumCost(int m, int n, vector<int>& horizontalCut, vector<int>& verticalCut) {
    ranges::sort(horizontalCut);
    ranges::sort(verticalCut);
    int64_t ans = 0;
    size_t i = 0, j = 0;
    for (size_t k = 0; k < m + n - 2; k++) {
      if (j == verticalCut.size() || (i < horizontalCut.size() && horizontalCut[i] < verticalCut[j])) {
        ans += horizontalCut[i] * (n - j);  // up/down
        ++i;
      } else {
        ans += verticalCut[j] * (m - i);  // left/right
        ++j;
      }
    }
    return ans;
  }

};

File: daily/Dec25.rs
impl Solution {
    pub fn minimum_cost(m: i32, n: i32, mut horizontal_cut: Vec<i32>, mut vertical_cut: Vec<i32>) -> i64 {
        let m = m as usize;
        let n = n as usize;
        horizontal_cut.sort_unstable();
        vertical_cut.sort_unstable();
        let mut ans = 0;
        let (mut i, mut j) = (0, 0);
        for _ in 0..m + n - 2 {
            if j == n - 1 || i < m - 1 && horizontal_cut[i] < vertical_cut[j] {
                ans += (horizontal_cut[i] * (n - j) as i32) as i64; // up/down
                i += 1;
            } else {
                ans += (vertical_cut[j] * (m - i) as i32) as i64;  // left/right
                j += 1;
            }
        }
        ans
    }
}
